api/envoy/config/core/v3/address.proto

```diff
@@ -164,6 +164,10 @@ message BindConfig {
   // :ref:`extra_source_addresses <envoy_v3_api_field_config.core.v3.BindConfig.extra_source_addresses>`
   repeated SocketAddress additional_source_addresses = 4
       [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
+
+  // The name of the custom local address selector. This must have been
+  // registered with Envoy. If this is empty, a default applies.
+  string local_address_selector_name = 6;
```

Comment:
Should we use `TypedExtensionConfig` https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/extension.proto#envoy-v3-api-msg-config-core-v3-typedextensionconfig just like other places?

It is more extensible if we have a selector which has its own config in the future

Reply:
cc @envoyproxy/api-shepherds ^

Reply:
cc @adisuissa @lizan

Reply:
Thank you so much for the review Adi! Do you have a suggestion of a good place to define the typed extension config for the default `UpstreamLocalAddressSelector` ?

Reply:
It can be under the `source/extension/upstream_local_address_selectors/` or `source/extension/upstreams/local_address_selectors`. just my two cents. cc @adisuissa